### PR TITLE
Bump golang 1.21 for v3.2.1 to avoid CVEs

### PR DESCRIPTION
**Pull request by zhengxiexie**

### CODE REVIEW

File: build/image/photon/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.19.10 as golang-build
+FROM golang:1.21.5 as golang-build
```
Comment (timdengyun) [inline build/image/photon/Dockerfile:1]: After upgrade golang to 1.21, you should be care is there is glibc issue.
https://github.com/vmware-tanzu/nsx-operator/pull/448
if yes, you should disable CGO.

**Comment by timdengyun** [comment]
https://github.com/vmware-tanzu/nsx-operator/blob/v3.2.1/.github/workflows/makefile.yml#L5
please also change github action from main branch to v3.2.1, so that we can see every golang update could make unit test run successfully.

File: build/image/photon/Dockerfile (hunk 1 above)
Comment (zhengxiexie) [reply build/image/photon/Dockerfile:1]: ok

**Comment by zhengxiexie** [comment]
> In addition, we should change https://github.com/vmware-tanzu/nsx-operator/blob/v3.2.1/.github/workflows/makefile.yml#L5 github action from main branch to v3.2.1, so that we can see every golang update could make unit test run successfully.

ok

**Review by timdengyun — APPROVED** [review]

**Pull request merged**